lib/api/core/models/repositories/profileRepository.js

```diff
@@ -7,37 +7,35 @@ var
   NotFoundError = require('kuzzle-common-objects').Errors.notFoundError,
   Repository = require('./repository');
 
+
+function extractRoleIds(policiesOrRoles) {
+  return policiesOrRoles.map(element => {
+    if (element.roleId) {
+      return element.roleId;
+    }
+    return element._id;
+  });
+}
+
 /**
+ * @class ProfileRepository
+ * @extends Repository
  * @param {Kuzzle} kuzzle
- * @returns {ProfileRepository}
+ * @constructor
  */
-module.exports = function (kuzzle) {
-  var extractRoleIds = policiesOrRoles => {
-    return policiesOrRoles.map(element => {
-      if (element.roleId) {
-        return element.roleId;
-      }
-      return element._id;
+function ProfileRepository (kuzzle) {
+  Repository.apply(this, arguments);
+  this.collection = 'profiles';
+  this.ObjectConstructor = Profile;
+  this.profiles = {};
+
+  ProfileRepository.prototype.init = function() {
```

Comment:
Overriding the prototype within the constructor is disturbing me.
 
In this particular case, we have a kind-of singleton and we could say that this is not that bad but still, conceptually, I am really not comfortable with designing class-level properties during the creation of an instance.